### PR TITLE
Injecting TPU_NAME environment variable

### PR DESCRIPTION
**Pull request by Edwinhr716**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
When requesting TPU resources, it injects the environment variable TPU_NAMES, which contains the name of the pod the TPU is in. This is needed as popular distributed runtime libraries such as Ray expect this variable to be set https://github.com/ray-project/ray/blob/ab27fe805c3fef4c8dc66bf5804f32cd522d1aad/python/ray/_private/accelerators/tpu.py#L247

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/utils/accelerators/tpu.go [hunk 1]
```diff
@@ -211,6 +216,10 @@ func AddTPUVariables(pod *corev1.Pod, size int) error {
 			Name:  TpuWorkerId,
 			Value: fmt.Sprint(tpuWorkerId),
 		},
+		corev1.EnvVar{
+			Name:  TpuName,
+			Value: fmt.Sprint(pod.Name),
```
Comment (liurupeng) [inline pkg/utils/accelerators/tpu.go:221]: I think this value is used by Ray to know which TPU host belongs to which slice? So if it's a multi-host slice, the value that be the same.  So one replica has one name, let's use leader pod's name here

Comment (liurupeng) [inline pkg/utils/accelerators/tpu.go:221]: the tpu_name here should be the identifier for a tpu slice. So for multi-host tpu slice, all hosts share the same tpu_name value. Let's use leader pod's name here.

Comment (liurupeng) [reply pkg/utils/accelerators/tpu.go:221]: actually this may cause the ray only to use one host to load the model which OOM happens on that host?

**Comment by BenTheElder** [comment]
/override pull-lws-test-e2e-main-1-27

see: https://kubernetes.slack.com/archives/C7J9RP96G/p1725908686722759 / https://kubernetes.slack.com/archives/C09QZ4DQB/p1725891002485159

**Comment by BenTheElder** [comment]
er, that command has to be  done by a github admin of this repo

**Comment by ahg-g** [comment]
btw, we can do this today as follows: https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/

File: pkg/utils/accelerators/tpu.go (hunk 1 above)
Comment (Edwinhr716) [reply pkg/utils/accelerators/tpu.go]: Updated to use leader pod's name

**Comment by BenTheElder** [comment]
maybe try closing and re-opening, that forces github to clear the statuses

**Pull request merged**